main.tf

```diff
@@ -54,7 +64,7 @@ locals {
   iam_allowed_ip_addresses              = var.enforce_allowed_ip_addresses == false ? "?${local.concatenated_allowed_ip_addresses}" : local.concatenated_allowed_ip_addresses
   iam_allowed_ip_addresses_control_mode = var.enforce_allowed_ip_addresses == false ? "MONITOR" : "RESTRICT"
   account_public_access                 = ibm_iam_access_group_account_settings.iam_access_group_account_settings.public_access_enabled
-  account_shell_settings_status         = ibm_cloud_shell_account_settings.cloud_shell_account_settings.enabled
+  account_shell_settings_status         = var.skip_cloud_shell_calls == true ? ibm_cloud_shell_account_settings.cloud_shell_account_settings[0].enabled : var.shell_settings_enabled
```

Comment:
you want this to default to null if the check is skipped so its not included as an output